x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/SecuritySearchOperationListener.java

```diff
@@ -44,25 +45,32 @@ public SecuritySearchOperationListener(SecurityContext securityContext, XPackLic
         this.auditTrailService = auditTrail;
     }
 
+    @Override
+    public void onNewReaderContext(ReaderContext readerContext) {
+        if (licenseState.isSecurityEnabled()) {
+            readerContext.putInContext(AuthenticationField.AUTHENTICATION_KEY, securityContext.getAuthentication());
+        }
+    }
+
     /**
-     * Adds the {@link Authentication} to the {@link ScrollContext}
+     * Adds the {@link Authentication} to the {@link ReaderContext}
      */
     @Override
-    public void onNewScrollContext(ScrollContext scrollContext) {
+    public void onNewScrollContext(ReaderContext readerContext) {
         if (licenseState.isSecurityEnabled()) {
-            scrollContext.putInContext(AuthenticationField.AUTHENTICATION_KEY, securityContext.getAuthentication());
+            readerContext.putInContext(AuthenticationField.AUTHENTICATION_KEY, securityContext.getAuthentication());
         }
     }
 
     /**
-     * Checks for the {@link ScrollContext} if it exists and compares the {@link Authentication}
-     * object from the scroll context with the current authentication context
+     * compares the {@link Authentication} object from the reader context with the current
+     * authentication context
      */
     @Override
-    public void validateSearchContext(SearchContext searchContext, TransportRequest request) {
+    public void validateSearchContext(ReaderContext readerContext, SearchContext searchContext, TransportRequest request) {
         if (licenseState.isSecurityEnabled()) {
             if (searchContext.scrollContext() != null) {
-                final Authentication originalAuth = searchContext.scrollContext().getFromContext(AuthenticationField.AUTHENTICATION_KEY);
+                final Authentication originalAuth = readerContext.getFromContext(AuthenticationField.AUTHENTICATION_KEY);
```

Comment:
I think we should validate the authentication of any reader context, not only the reader contexts with scroll?

Reply:
Oups, I pushed a change to address this:
https://github.com/elastic/elasticsearch/pull/54966/commits/0ec84abf64beeae6c5dbb35217306e447ca725d8